modules/govuk_jenkins/files/var/lib/jenkins/groovy_scripts/govuk_jenkinslib.groovy

```diff
@@ -367,4 +367,156 @@ def withStatsdTiming(key, fn) {
   sh 'echo "ci.' + project_name + '.' + key + ':' + runtime + '|ms" | nc -w 1 -u localhost 8125'
 }
 
+/**
+ * Build the project
+ * @param sassLint Whether or not to run the SASS linter
+ */
+def buildProject(sassLint = true) {
+  repoName = JOB_NAME.split('/')[0]
+
+  properties([
+    buildDiscarder(
+      logRotator(
+        numToKeepStr: '50')
+      ),
+    [$class: 'RebuildSettings', autoRebuild: false, rebuildDisabled: false],
+    [$class: 'ParametersDefinitionProperty',
+      parameterDefinitions: [
+        [$class: 'BooleanParameterDefinition',
+          name: 'IS_SCHEMA_TEST',
+          defaultValue: false,
+          description: 'Identifies whether this build is being triggered to test a change to the content schemas'],
+        [$class: 'StringParameterDefinition',
+          name: 'SCHEMA_BRANCH',
+          defaultValue: 'deployed-to-production',
+          description: 'The branch of govuk-content-schemas to test against']]
+    ],
+  ])
+
+  try {
+    if (!isAllowedBranchBuild(env.BRANCH_NAME)) {
+      return
+    }
+
+    stage("Checkout") {
+      checkoutFromGitHubWithSSH(repoName)
+    }
+
+    stage("Clean up workspace") {
+      cleanupGit()
+    }
+
+    stage("Merge master") {
+      mergeMasterBranch()
+    }
+
+    stage("Configure environment") {
+      setEnvar("RAILS_ENV", "test")
+      setEnvar("RACK_ENV", "test")
+    }
+
+    stage("Set up content schema dependency") {
+      contentSchemaDependency(params.SCHEMA_BRANCH)
+      setEnvar("GOVUK_CONTENT_SCHEMAS_PATH", "tmp/govuk-content-schemas")
+    }
+
+    stage("bundle install") {
+      if (isGem()) {
+        bundleGem()
+      } else {
+        bundleApp()
+      }
+    }
+
+    if (hasLint()) {
+      stage("Lint Ruby") {
+        rubyLinter("app lib spec test")
+      }
+    } else {
+      echo "WARNING: You do not have Ruby linting turned on. Please install govuk-lint and enable."
+    }
+
+    if (hasAssets() && sassLint) {
+      stage("Lint SASS") {
+        sassLinter()
+      }
+    } else {
+      echo "WARNING: You do not have SASS linting turned on. Please install govuk-lint and enable."
+    }
+
+    if (hasDatabase()) {
+      stage("Set up the database") {
+        runRakeTask("db:drop db:create db:schema:load")
+      }
+    }
+
+    stage("Run tests") {
+      // Prevent a project's tests from running in parallel on the same node
+      lock("$repoName-$NODE_NAME-test") {
+        runTests()
+      }
+    }
+
+    if (hasAssets()) {
+      stage("Precompile assets") {
+        precompileAssets()
+      }
+    }
+
+    if (isGem()) {
```

Comment:
We should check that this is the master branch before running any of this. It's a redundant check because `deployIntegration`, etc. also check the branch, but it stops "Deploy to integration" from appearing in branch build summaries.